plugins/akamai.js

```diff
@@ -0,0 +1,126 @@
+#!/usr/bin/env node
+
+var EdgeGrid = require('edgegrid');
+var dnszone = process.env.DOMAIN;
+
+if (dnszone == undefined){
+  console.error("No DOMAIN specified");
+  process.exit(1);
+}
+
+var stdin = process.openStdin();
+var spf_records = [];
+
+stdin.on('data', function(data){
+  spf_records.push(parseSpfItem(data));
+});
+
+stdin.on('end', function(){
+  console.log('done!');
+  console.log(JSON.stringify(spf_records));
+  getZoneFile(spf_records, function(error, zonefile){
+    if (error){
+      console.error(error);
+      process.exit(1);
+    }
+    updateZoneFile(zonefile, spf_records);
+    postZoneFile(zonefile);
+  });
+});
+
+
+function parseSpfItem(data){
+  var spfpartial = data.toString().split("^");
+  return {
+    target: JSON.parse(spfpartial[1]),
+    active: true,
+    name: getSpfItemName(spfpartial[0]),
+    ttl: 300
+  }
+}
+
+function getSpfItemName(data){
+  var itemName = data.replace("." + dnszone, '');
+  if (data === dnszone) itemName = null;
+  return itemName;
+}
+
+function getZoneFile(spf_records, cb){
+  var eg = new EdgeGrid({
+    path: '.edgerc',
+    section: 'default'
+  });
+
+  eg.auth({
+    path: '/config-dns/v1/zones/' + dnszone,
+    method: 'GET',
+    headers: {}
+  });
+
+  eg.send(function (error, response, body) {
+    exitOnError(error, response, body);
+    cb(error,  JSON.parse(body));
```

Comment:
I'd say it's not safe for the asynchronous calls. If the `process.exit(1)` would be delayed, it could trigger callback execution.